lib/cores.js

```diff
@@ -108,36 +109,45 @@ async function getConnectedCoreIPsAsync(coreIPList, coreConnectionCount) {
   }
 }
 
-async function coreRequestAsync(options, coreIP, retryCount = 3) {
+async function coreRequestAsync(options, coreIP, retryCount = 3, timeout = 500) {
   options.headers['X-Node-Version'] = version
   options.uri = `http://${coreIP}${options.uriPath}`
 
   let response
   if (retryCount <= 0) {
-    response = await rp(options)
+    try {
+      response = await rp(options)
+    } catch (error) {
+      if (error.statusCode === 402 || error.status === 402) return error.response
+      else throw error
+    }
   } else {
     await retry(
       async bail => {
         try {
           response = await rp(options)
         } catch (error) {
-          // If no response was received or there is a status code >= 500, then we should retry the call, throw an error
-          if (!error.statusCode || error.statusCode >= 500) throw error
+          // If no response was received or there is a status code >= 500 or payment is still pending/held,
+          // then we should retry the call, throw an error
+          if (!error.statusCode || error.statusCode >= 500 || error.statusCode === 402) throw error
```

Comment:
If a submission is made before the lightning payment to the hodl invoice has been processed by the lightning network, then you'll get a 402 back in response. We want to keep trying in this case.